resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/kubernetes/submit/submitsteps/initcontainer/SubmittedResourcesInitContainerStepSuite.scala

```diff
@@ -0,0 +1,173 @@
+
+  before {
+    MockitoAnnotations.initMocks(this)
+    when(submittedDependencyUploader.uploadJars()).thenReturn(
+      SubmittedResourceIdAndSecret(JARS_RESOURCE_ID, JARS_SECRET)
+    )
+    when(submittedDependencyUploader.uploadFiles()).thenReturn(
+      SubmittedResourceIdAndSecret(FILES_RESOURCE_ID, FILES_SECRET)
+    )
+    when(submittedResourcesSecretPlugin.mountResourceStagingServerSecretIntoInitContainer(
+      any[Container])).thenReturn(
+        new ContainerBuilder().withName("mountedSecret").build())
+    when(submittedResourcesSecretPlugin.addResourceStagingServerSecretVolumeToPod(
+      any[Pod])).thenReturn(
+      new PodBuilder()
+        .withNewMetadata()
+        .addToLabels("mountedSecret", "true")
+        .endMetadata()
+        .withNewSpec().endSpec()
+        .build())
+  }
+  test ("testing vanilla prepareInitContainer on resources and properties") {
+    val submittedResourceStep = new SubmittedResourcesInitContainerStep(
+      RESOURCE_SECRET_NAME,
+      STAGING_SERVER_URI,
+      SECRET_MOUNT_PATH,
+      false,
+      None,
+      None,
+      None,
+      None,
+      submittedDependencyUploader,
+      submittedResourcesSecretPlugin
+    )
+    val returnedInitContainer =
+      submittedResourceStep.prepareInitContainer(InitContainerSpec(
+        Map.empty[String, String],
+        Map.empty[String, String],
+        new Container(),
+        new Container(),
+        new Pod(),
+        Seq.empty[HasMetadata]))
+    assert(returnedInitContainer.initContainer.getName === "mountedSecret")
+    assert(returnedInitContainer.podToInitialize.getMetadata.getLabels.asScala
+      === Map("mountedSecret" -> "true"))
+    assert(returnedInitContainer.initContainerDependentResources.length == 1)
+    val secret = returnedInitContainer.initContainerDependentResources.head.asInstanceOf[Secret]
+    assert(secret.getData === RSS_SECRET)
+    assert(secret.getMetadata.getName == RESOURCE_SECRET_NAME)
+    val expectedinitContainerProperties = Map(
+      RESOURCE_STAGING_SERVER_URI.key -> STAGING_SERVER_URI,
+      INIT_CONTAINER_DOWNLOAD_JARS_RESOURCE_IDENTIFIER.key -> JARS_RESOURCE_ID,
+      INIT_CONTAINER_DOWNLOAD_JARS_SECRET_LOCATION.key ->
+        s"$SECRET_MOUNT_PATH/$INIT_CONTAINER_SUBMITTED_JARS_SECRET_KEY",
+      INIT_CONTAINER_DOWNLOAD_FILES_RESOURCE_IDENTIFIER.key -> FILES_RESOURCE_ID,
+      INIT_CONTAINER_DOWNLOAD_FILES_SECRET_LOCATION.key ->
+        s"$SECRET_MOUNT_PATH/$INIT_CONTAINER_SUBMITTED_FILES_SECRET_KEY",
+      RESOURCE_STAGING_SERVER_SSL_ENABLED.key -> false.toString)
+    assert(returnedInitContainer.initContainerProperties === expectedinitContainerProperties)
+    assert(returnedInitContainer.additionalDriverSparkConf ===
+      Map(
+        EXECUTOR_INIT_CONTAINER_SECRET.key -> RESOURCE_SECRET_NAME,
+        EXECUTOR_INIT_CONTAINER_SECRET_MOUNT_DIR.key -> SECRET_MOUNT_PATH))
+  }
+
+  test ("testing prepareInitContainer w/ CERT and TrustStore Files w/o SSL") {
```

Comment:
This test checks for when we're sending files from the submitter's local disk to mount as a secret. But the user is also allowed to specify these trustStore paths as `local://` paths if they're injecting their certificates into the init-container in a different way. We should have a unit test for that.